com.civicrm.hrjobroles/gulpfile.js

```diff
@@ -0,0 +1,111 @@
+var gulp = require('gulp');
+var clean = require('gulp-clean');
+var rename = require('gulp-rename');
+var replace = require('gulp-replace');
+var bulk = require('gulp-sass-bulk-import');
+var sass = require('gulp-sass');
+var karma = require('karma');
+var exec = require('child_process').exec;
+var path = require('path');
+var fs = require('fs');
+
+gulp.task('sass', function (done) {
+    // The app style relies on compass's gems, so we need to rely on it
+    // for the time being
+    exec('compass watch', function (_, stdout, stderr) {
+        console.log(stdout);
+        done();
+    });
+});
+
+gulp.task('requirejs-bundle', function (done) {
+    exec('r.js -o js/build.js', function (_, stdout, stderr) {
+        done();
+    });
+});
+
+gulp.task('watch', function () {
+    gulp.watch('scss/**/*.scss', ['sass']);
+    gulp.watch('js/src/**/*.js', ['requirejs-bundle']).on('change', function (file) {
+        try { test.for(file.path); } catch (ex) { test.all(); };
+    });
+    gulp.watch(['js/test/**/*.js', '!js/test/mocks/**/*.js', '!js/test/test-main.js']).on('change', function (file) {
+        test.single(file.path);
+    });
+});
+
+gulp.task('test', function (done) {
+    test.all();
+});
+
+gulp.task('default', ['requirejs-bundle', 'sass', 'test', 'watch']);
+
+
+
+var test = (function () {
+
+    /**
+     * Runs the karma server which does a single run of the test/s
+     *
+     * @param {string} configFile - The full path to the karma config file
+     * @param {Function} cb - The callback to call when the server closes
+     */
+    function runServer(configFile, cb) {
+        new karma.Server({
+            configFile: __dirname + '/js/' + configFile,
+            singleRun: true
+        }, function () {
+            cb && cb();
+        }).start();
+    }
+
+    return {
+
+        /**
+         * Runs all the tests
+         */
+        all: function () {
+            runServer('karma.conf.js');
+        },
+
+        /**
+         * Runs the tests for a specific source file
+         *
+         * Looks for a test file (*_test.js) in `test/`, using the same path
+         * of the source file in `src/job-roles/`
+         *   i.e. src/job-roles/models/model.js -> test/models/model_test.js
+         *
+         * @throw {Error}
```

Comment:
I think this is also misleading since function does not throw any errors.
If fixed - approved.